src/linter/block.go

```diff
@@ -1604,18 +1604,42 @@ func (b *BlockWalker) handleAssignReference(a *ir.AssignReference) bool {
 	return false
 }
 
-func (b *BlockWalker) handleAssignList(items []*ir.ArrayItemExpr, info meta.ClassInfo, isShape bool) {
-	if isShape {
-		for i, item := range items {
-			prop, ok := info.Properties[fmt.Sprint(i)]
-			var tp meta.TypesMap
-			if !ok {
-				tp = meta.NewTypesMap("unknown_from_list")
-			} else {
-				tp = prop.Typ
+func (b *BlockWalker) handleAssignShapeToList(items []*ir.ArrayItemExpr, info meta.ClassInfo) {
+	for i, item := range items {
+		var prop meta.PropertyInfo
+		var ok bool
+
+		if item.Key != nil {
+			var key string
+			switch keyNode := item.Key.(type) {
+			case *ir.String:
+				key = unquote(keyNode.Value)
+			case *ir.Lnumber:
+				key = keyNode.Value
+			case *ir.Dnumber:
+				key = keyNode.Value
+			default:
+				key = "unhandled_type_of_key"
```

Comment:
I would keep the empty string and avoid lookup for `unhandled_type_of_key`.
It's also suspicious if there is a `unhandled_type_of_key` key in a shape. ;)